release

```diff
@@ -0,0 +1,190 @@
+ * - Generate the list of commands to run
+ *
+ * Usage:
+ * 1. run the script without argument. ie. `./release`
+ * 2. copy the output and keep it for later
+ * 3. open a PR for the changes generated by the script => review => merge the PR
+ * 4. ⚠️ ASSERT the splitter ran successfully!! ⚠️
+ * 5. run the commands generated by the step 2.
+ *
+ * @internal
+ */
+
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Finder\Finder;
+
+if (!file_exists(__DIR__ . '/vendor/autoload.php')) {
+    echo "Run `composer install` before you run the `generate` script.\n";
+    exit(1);
+}
+
+require __DIR__ . '/vendor/autoload.php';
+
+function dumpFile(string $filename, $content): void
+{
+    static $filesystem;
+    if (!$filesystem) {
+        $filesystem = new Filesystem();
+    }
+
+    $filesystem->dumpFile($filename, $content);
+}
+function getChangeLogs(): iterable
+{
+    $finder = (new Finder())
+        ->in(__DIR__ . '/src')
+        ->name('CHANGELOG.md');
+
+    /** @var \SplFileInfo $file */
+    foreach ($finder as $file) {
+        yield $file->getRealPath();
+    }
+}
+
+function getNextVersion(string $changelogPath): ?string
+{
+    $lines = explode("\n", file_get_contents($changelogPath));
+    $started = false;
+    $level = 3; // 0=major, 1=minor, 2=patch
+    $lastVersion = null;
+    if ('## NOT RELEASED' !== $lines[2]) {
+        throw new \Exception(sprintf('The "%s" changelog file does not contain the "## NOT RELEASED" section', $changelogPath));
+    }
+    foreach ($lines as $line) {
+        if (!$started) {
+            if ('## NOT RELEASED' === $line) {
+                $started = true;
+            }
+
+            continue;
+        }
+        if (!$line || '#' !== $line[0]) {
+            continue;
+        }
+        if ('### BC-BREAK' === $line || '### Removed' === $line) {
+            $level = 0;
+
+            break;
+        }
+        if ('### Added' === $line || '### Deprecated' === $line) {
+            $level = min($level, 1);
+
+            continue;
+        }
+        if ('### Changed' === $line || '### Fixed' === $line || '### Security' === $line) {
+            $level = min($level, 2);
+
+            continue;
+        }
+
+        if (0 === strpos($line, '## ')) {
+            $lastVersion = substr($line, 3);
+
+            break;
+        }
+    }
+    if (!$started) {
+        throw new \Exception(sprintf('The "%s" changelog file does not contain the "## NOT RELEASED" section', $changelogPath));
+    }
+    if ($level > 2) {
+        return null;
+    }
+    if (!$lastVersion) {
+        throw new \Exception(sprintf('The "%s" changelog file does not contain a previous version', $changelogPath));
+    }
+    if (!preg_match('/^\d+\.\d+\.\d+$/', $lastVersion)) {
+        throw new \Exception(sprintf('The "%s" changelog file does not contain a valid previous version', $changelogPath));
+    }
+
+    $parts = explode('.', $lastVersion);
+    if (0 !== (int) $parts[0]) {
```

Comment:
```suggestion
    if (0 === (int) $parts[0]) {
```

Reply:
?

Reply:
this code handles `0.x` version which behaves differently:
- BC-BREAK in minor
- Feature in patch

That's the reason for `$level = min(2, $level + 1);` (the number that should be increased moves to the right)